pkg/test/context.go

```diff
@@ -76,6 +78,11 @@ func (ctx *TestCtx) GetID() string {
 }
 
 func (ctx *TestCtx) Cleanup() {
+	if ctx.t != nil {
+		if ctx.t.Failed() && !ctx.cleanupOnError {
+			return
```

Comment:
WDYT about add an info log here such as follows?

```suggestion
                         // Also, could we log the error here? 
                         log.Info("Skipping cleanup function since --skip-cleanup-error is true and an error was faced)  
			return
```